### ISSUE TITLE
In the. vue file, when lang="tsx", an error is reported when writing JSX syntax parsing

### ISSUE THREAD

**Issue by plusface** [vuejs/eslint-plugin-vue#2274]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.47.0
- **eslint-plugin-vue version:** 9.17.0
- **Node version:** 18.16.0
- **Operating System:** Windows

**Please show your full configuration:**

```json5
{ 
  "extends": [ 
    "plugin:vue/vue3-recommended", 
    "plugin:vue/vue3-strongly-recommended", 
    "plugin:vue/vue3-essential" 
  ], 
  "parser": "vue-eslint-parser", 
  "plugins": [ 
    "@typescript-eslint" 
  ], 
  "parserOptions": { 
    "ecmaFeatures": { 
      "jsx": true 
    }, 
    "parser": "@typescript-eslint/parser", 
    "project": "./tsconfig.json", 
    "extraFileExtensions": [ 
      ".vue" 
    ], 
    "vueFeatures": { 
      "ecmaFeatures": { 
        "jsx": true 
      } 
    } 
  }, 
  "root": true, 
  "rules": { 
    "@typescript-eslint/no-unsafe-argument": "warn" 
  } 
} 
```

**What did you do?**	

```vue
<template>
  <div></div>
</template>
<script lang="tsx" setup>
const b: number = 1
const a = <div></div>
console.log('[ b ]', b)
</script>
```

**What did you expect to happen?**
There should be no errors in theory

**What actually happened?**

Reported a parsing error：`Parsing error: Cannot set property index of Error which has only a getter`.
By commented code and observing，Parsing is normal when it does not include jsx syntax.
**Repository to reproduce this issue**

https://github.com/plusface/eslint-vue-tsx-bug.git
I saw the same error on `eslint plugin vue demo`：https://ota-meshi.github.io/eslint-plugin-vue-demo/#eJxVzk0KwjAQBeCrDLOpgpp9SYsHySamQwnkZ0imRRHvbkoVdDOL98F780SXJ8IetVDkYIVGkwD05NdRq+2apNWP6eqKZ4FKsjAEm+bBoNS7wYYupypg+7TEGxUYoOu+ocuRW/BXvEkOdAl5PtjjNrSXN8ITsi2VSnvtKg+mXc5Ug0+iPvZ6A/KuQ1M=

**Comment by PurpleTape** [comment]
I've come across something similar. You can use `@vue/eslint-config-airbnb-with-typescript` package or copy the configuration of `parserOptions` from there

Source: https://github.com/vuejs/eslint-config-airbnb/tree/main/packages/eslint-config-airbnb-with-typescript#using-tsxjsx-in-vue-files-strongly-not-recommended

**Comment by plusface** [comment]
> 我也遇到过类似的事情。您可以使用包或从那里`@vue/eslint-config-airbnb-with-typescript`复制配置`parserOptions`
> 
> 来源：https://github.com/vuejs/eslint-config-airbnb/tree/main/packages/eslint-config-airbnb-with-typescript#using-tsxjsx-in-vue-files-strongly-not-recommended

thank you!

But I took a look at `@vue/eslint-config-airbnb-with-typescript` package, it seems to set `project` in `parserOptions` to `null`, before that, I have tried `project` field When no value is specified, jsx syntax can be parsed normally. But if rules such as [`@typescript-eslint/no-unsafe-argument`](https://typescript-eslint.io/rules/no-unsafe-argument/) are used, the official requirement must be to set the `project` field to `true` or the specified path of `tsconfig.json`, so `@typescript-eslint/ The no-unsafe-argument` rule conflicts with not specifying the `project` field.

At present, there is really no good solution, so I am not going to use `@typescript-eslint/no-unsafe-argument` Such rules.

**Comment by plusface** [comment]
> I've come across something similar. You can use `@vue/eslint-config-airbnb-with-typescript` package or copy the configuration of `parserOptions` from there
> 
> Source: https://github.com/vuejs/eslint-config-airbnb/tree/main/packages/eslint-config-airbnb-with-typescript#using-tsxjsx-in-vue-files-strongly-not-recommended

Sorry, I looked at the official documentation again, and this situation was mentioned. There is this limitation, so the official strongly recommends not using `tsx/jsx` in `.vue` files. I think I got it, thank you.

**Comment by sgpinkus** [comment]
Why is this closed? What's the solution for parsing .vue file with tsx in them?

I'm not sure if it works at all but in any case @vue/eslint-config-airbnb-with-typescript only works with eslint upto v8, not latest v9.
